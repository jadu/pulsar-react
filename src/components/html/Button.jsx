import React from 'react';
import classnames from 'classnames';

export default class Button extends React.Component {

  static defaultProps = {
    className: 'btn'
  };

  render() {
    let {
      children,
      className,
      danger,
      inverse,
      naked,
      primary,
      success,
      warning,
      ...props
    } = this.props;

    let variantClasses = classnames(className, {
      'btn--danger': danger,
      'btn--inverse': inverse,
      'btn--naked': naked,
      'btn--primary': primary,
      'btn--success': success,
      'btn--warning': warning
    });

    return <button className={variantClasses} {...props}>{children}</button>;
  }
}
