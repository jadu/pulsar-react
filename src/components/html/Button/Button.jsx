import React from 'react';
import classnames from 'classnames';

export default class Button extends React.Component {

  static defaultProps = {
    className: 'btn',
    tag: 'button',
    type: ''
  };

  render() {
    let {
      children,
      className,
      danger,
      href,
      info,
      inverse,
      naked,
      nodeRef,
      outline,
      primary,
      small,
      tag,
      type,
      success,
      warning,
      white,
      ...props
    } = this.props;

    let ButtonTag = href ? 'a' : tag;

    let variantClasses = classnames(className, {
      'btn--danger': type === 'danger',
      'btn--info': type === 'info',
      'btn--inverse': type === 'inverse',
      'btn--naked': type === 'naked',
      'btn--outline': outline,
      'btn--primary': type === 'primary',
      'btn--success': type === 'success',
      'btn--small': small,
      'btn--warning': type === 'warning',
      'btn--white': type === 'white'
    });

    return (
      <React.Fragment>
        <ButtonTag 
          className={variantClasses}
          href={href ? href : null}
          ref={nodeRef}
          {...props}
        >
          {children}
        </ButtonTag>
      </React.Fragment>
    );
  }
}
