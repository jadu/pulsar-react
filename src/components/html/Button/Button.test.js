'use strict';

import React from 'react';
import { shallow, render } from 'enzyme';

import Button from './Button';

it('renders the base button', () => {
  const tree = render(
    <Button>foo</Button>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the primary variation', () => {
  const tree = render(
    <Button primary>foo</Button>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the success variation', () => {
  const tree = render(
    <Button success>foo</Button>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the warning variation', () => {
  const tree = render(
    <Button warning>foo</Button>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the danger variation', () => {
  const tree = render(
    <Button danger>foo</Button>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the inverse variation', () => {
  const tree = render(
    <Button inverse>foo</Button>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the outline variation', () => {
  const tree = render(
    <Button outline>foo</Button>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the naked variation', () => {
  const tree = render(
    <Button naked>foo</Button>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the small variation', () => {
  const tree = render(
    <Button small>foo</Button>
  );
  expect(tree).toMatchSnapshot();
});

it('renders multiple variations', () => {
  const tree = render(
    <Button primary outline small>foo</Button>
  );
  expect(tree).toMatchSnapshot();
});

it('renders as a link if the href attribute is supplied', () => {
  const tree = render(
    <Button href="#bar">foo</Button>
  );
  expect(tree).toMatchSnapshot();
});