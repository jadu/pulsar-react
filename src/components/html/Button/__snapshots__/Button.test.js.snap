// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as a link if the href attribute is supplied 1`] = `
<a
  class="btn"
  href="#bar"
>
  foo
</a>
`;

exports[`renders multiple variations 1`] = `
<button
  class="btn btn--outline btn--primary btn--small"
>
  foo
</button>
`;

exports[`renders the base button 1`] = `
<button
  class="btn"
>
  foo
</button>
`;

exports[`renders the danger variation 1`] = `
<button
  class="btn btn--danger"
>
  foo
</button>
`;

exports[`renders the inverse variation 1`] = `
<button
  class="btn btn--inverse"
>
  foo
</button>
`;

exports[`renders the naked variation 1`] = `
<button
  class="btn btn--naked"
>
  foo
</button>
`;

exports[`renders the outline variation 1`] = `
<button
  class="btn btn--outline"
>
  foo
</button>
`;

exports[`renders the primary variation 1`] = `
<button
  class="btn btn--primary"
>
  foo
</button>
`;

exports[`renders the small variation 1`] = `
<button
  class="btn btn--small"
>
  foo
</button>
`;

exports[`renders the success variation 1`] = `
<button
  class="btn btn--success"
>
  foo
</button>
`;

exports[`renders the warning variation 1`] = `
<button
  class="btn btn--warning"
>
  foo
</button>
`;
